src/core/interfaces.ts

```diff
@@ -88,11 +88,22 @@ export type Bounds = {
 };
 
 /**
+ * Client bounds
+ */
+export interface IEntityBounds {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+/**
  * An object representing a data-backed visual entity inside a Component.
  */
 export interface IEntity<C extends Component> {
+  bounds?: IEntityBounds; // TODO make non-optional
```

Comment:
This API feels very specific to rects and implementing tooltips for rects. Could we instead expose the resolved attrs associated with this Entity?